core/core.go

```diff
@@ -367,6 +367,9 @@ func (n *IpfsNode) SetupOfflineRouting() error {
 	}
 
 	n.Routing = offroute.NewOfflineRouter(n.Repo.Datastore(), n.PrivateKey)
+
+	n.Namesys = namesys.NewNameSystem(n.Routing)
```

Comment:
(another question entirely is what _ought to be_ and what _should be now_ -- this may be out of scope in this PR).